lua/starfall/libs_sh/vr.lua

```diff
@@ -0,0 +1,413 @@
+
+local checkluatype = SF.CheckLuaType
+
+--- VRMod Library 
+-- Addon and module: https://steamcommunity.com/sharedfiles/filedetails/?id=1678408548
+-- Follow install instructions on the addon's page.
+-- @name vr
+-- @class library
+-- @libtbl vr_library
+SF.RegisterLibrary("vr")
+
+return function(instance)
+
+if vrmod then --only add these functions if vrmod is installed
+
+local vr_library = instance.Libraries.vr
+
+local owrap, ounwrap = instance.WrapObject, instance.UnwrapObject
+local ents_methods, ent_meta, ewrap, eunwrap = instance.Types.Entity.Methods, instance.Types.Entity, instance.Types.Entity.Wrap, instance.Types.Entity.Unwrap
+local ang_meta, awrap, aunwrap = instance.Types.Angle, instance.Types.Angle.Wrap, instance.Types.Angle.Unwrap
+local vec_meta, vwrap, vunwrap = instance.Types.Vector, instance.Types.Vector.Wrap, instance.Types.Vector.Unwrap
+local plywrap, plyunwrap = instance.Types.Player.Wrap, instance.Types.Player.Unwrap
+
+--- Called when a player enters VR
+-- @name VRStart
+-- @class hook
+-- @param ply Player entering VR
+SF.hookAdd("VRMod_Start", "vrstart")
+
+--- Called when a player exits VR
+-- @name VRExit
+-- @class hook
+-- @param ply Player exiting VR
+SF.hookAdd("VRMod_Exit", "vrexit")
+
+
+--- Checks wether the player is in VR
+-- @class function
+-- @param target player to check
+-- @return boolean true if player is in VR
+function vr_library.isPlayerInVR(ply)
+	local ply = plyunwrap(ply)
+	checkluatype(ply, TYPE_ENTITY) 
```

Comment:
none of the `checkluatype(ply, TYPE_ENTITY)` are needed. They can be removed.

Reply:
You should instead use
```lua
local getply
instance:AddHook("initialize", function()
	getply = instance.Types.Player.GetPlayer
end)
```
`getply(ply)` instead of `plyunwrap(ply)`